bootloader/assets/assets.go

```diff
@@ -0,0 +1,48 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package assets
+
+import (
+	"fmt"
+)
+
+var registeredAssets = map[string][]byte{}
+
+// RegisterAsset registers a boot asset under the given name.
+func RegisterAsset(name string, data []byte) {
```

Comment:
I don't think this should be exported, since if other packages import this pkg to register their assets, consumers of that asset will need to remember to import both that package and this one, and we will have the same problem I found with the squashfs format refactoring in https://github.com/snapcore/snapd/pull/8727 which is not great in general. 

To combat doing that accidentally, we should just make this not exported to prevent accidental usages of this from other packages. If we have tests that need to do this, then we can make it a Mock* function, etc.